api/resource/resource.go

```diff
@@ -351,6 +375,18 @@ func (r *Resource) CurId() resid.ResId {
 		r.GetGvk(), r.GetName(), r.GetNamespace())
 }
 
+// setRefBy sets the resource's RefBy annotation
+func (r *Resource) setRefBy(ids []resid.ResId) {
+	annotations := r.GetAnnotations()
+	if len(annotations) > 0 {
+		delete(annotations, utils.BuildAnnotationsRefBy)
+		r.SetAnnotations(annotations)
+	}
+	for _, id := range ids {
```

Comment:
Looks like we are repeatedly getting, mutating and setting the annotations here. Could we instead make a new helper like `r.replaceCsvAnnotation(name, values...)` to make this more straightforward/efficient?

Reply:
`mergeStringMapsWithBuildAnnotations` makes `r.setRefBy` unnecessary so I removed it